Training

mirisuzanne opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&cute)
<!-- Swap `CHANGE_ME` for a random string to get a different image -->


## Description

- update upcoming announced talks
- update past talk video
- update how talks are displayed: https://trello.com/c/zEoA48al/134-how-are-resources-sorted-does-it-make-sense


## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

This touches a lot of pages:

- training removed from services
- talks and workshops removed from blog
- talks page added to top nav
- talks page has upcoming events grouped by bird
- talks page has list of talks and workshops with latest two events shown for each
- bird pages show upcoming talks

mirisuzanne commented:
@jgerigmeyer I broke something in the tests here, if you can take a look at that.

jgerigmeyer commented on content/_includes/page/meta.njk:
```diff
@@ -37,7 +37,7 @@
           start=page.date,
           end=end,
-          format='range' if oss or client else 'long'
+          format='range' if oss or client or event else 'long'
```
I added this `event` data to talks/workshops so that we could replace the specific date with `since` -- but feel free to adjust if there's a better way to do this.

jgerigmeyer commented on content/_data/taxonomy.yaml:
```diff
@@ -1,10 +1,4 @@
 post:
-  - tag: Talk
-    icon: talk
-    img: true
-  - tag: Workshop
-    icon: workshop
-    img: true
```
I don't fully understand the logic, but these types of posts still appear in lists on e.g. oss-detail pages (https://deploy-preview-290--oddleventy.netlify.app/accoutrement/). But now they're missing little corner icons, and they link to the /tags/ page.

SondraE replied:
Like we talked about in the meeting this morning, I think it would be fine (even useful?) for talks/workshops to show up as related posts for Birds and OSS projects and in tag lists. Looks like they lost their images and icons, though.

mirisuzanne replied:
Should be fixed now

jgerigmeyer commented on content/_data/taxonomy.yaml:
```diff
@@ -1,21 +1,19 @@
     img: true
   - tag: Article
     icon: news
+    img: true
```
Removed in https://github.com/oddbird/oddleventy/commit/5e03c100a39df052b9490aa1d197128e128e083e

jgerigmeyer commented on src/filters/events.cjs:
```diff
@@ -82,38 +119,35 @@ const getEvents = (collection) => {
+    type: collection of events
 */
-const getFuture = (events) => events.filter(isFuture);
+const birdEvents = (events) => {
```
This new filter needs tests. (I can do that tomorrow if you'd prefer not to.)

mirisuzanne replied:
Yeah, if you have time that would be great. I can address the other notes here.

mirisuzanne commented on content/birds/miriam.md:
```diff
@@ -4,13 +4,24 @@ sub: Co-founder & product lead
   codepen: MiriamSuzanne
   stackoverflow: 1930386
 events:
+  - venue: Front Conference
+    date: 2023-08-31
+    end: 2023-09-01
+    adr: Zurich, Switzerland
+    url: https://frontconference.com/
```
Yeah, especially since some events get announced before we determine what talk will be happening at that event.

SondraE reviewed (approved):
I left a reply to one of Jonny's comments. Basically, I'm wondering if we want past talks/workshops to show up on Bird detail pages with the rest of the related posts just like they show up in related posts for OSS projects and on tag pages.

Also got this message from Catherine Holmes (the "Sponsors Co-Chair" for DjangoCon) about Ed's workshop. "[Ed's] tutorial has been the most popular this year!" I wonder if there's a place for that quote on the Talks & Workshops page.

mirisuzanne commented:
@SondraE done, see what you think?

SondraE commented:
@mirisuzanne I like having that quote there about Ed! Does Sarah's quote about your speaking still appear somewhere else, like on a specific talk page?

mirisuzanne commented:
@SondraE yes, that quote should still appear on both the agile-systems talk, and the vue conf retrospective.

SondraE commented on content/_includes/post.macros.njk:
```diff
@@ -243,7 +244,7 @@ params:
       {% if post.data.events %}
         <div class="p-summary">
-          {{ eventpreview(post.data.events, links=false) }}
+          {{ talks.list(post | pageEvents, compact=true) }}
```
For the cards, we are now using the same pattern we have on talk detail pages. The Upcoming talks have a bold date and Past talk have a regular font weight date. If that's not enough, we could add the labels Upcoming and Past in both places.

<img width="694" alt="Screen Shot 2022-12-12 at 10 36 24 AM" src="https://user-images.githubusercontent.com/8933115/207114747-87a92fb1-4c5a-40a0-9e6f-cb549ec9c154.png">
<img width="1440" alt="Screen Shot 2022-12-12 at 10 36 35 AM" src="https://user-images.githubusercontent.com/8933115/207114753-0c2fcd2e-e51e-4438-80fb-0522c6f8e47b.png">

mirisuzanne replied:
I think the final change we might want to make here is removing the unnecessary indent?

stacyk replied:
> For the cards, we are now using the same pattern we have on talk detail pages. The Upcoming talks have a bold date and Past talk have a regular font weight date. If that's not enough, we could add the labels Upcoming and Past in both places.

It isn't obvious at first why one date is bolded and other one isn't, but maybe that is OK? I do think the labels were instantly more clear. Did we want to save a bit of vertical space?

dvdherron replied:
Maybe the upcoming event dates could be links to the event site instead of being bold?

SondraE replied:
I originally removed the Upcoming and Past labels from events in the compact view to save space, but we can certainly add them back in for clarity. Saving space isn't super important. (And remove the bold text if it looks like a bug! 🐞 )

SondraE reviewed (approved):
lgtm! I'm excited for this update

stacyk commented on content/talks.njk:
```diff
@@ -0,0 +1,57 @@
+      <header class="training-header">
+        {{ post.banner(item) }}
+        {{ post.subtitle(item.data.sub) }}
+        {{ post.byline(item, collections, dateformat='range') }}
```
After reviewing the comments in the XD file, I am assuming we went this direction in order to not show the "published date" as a post would in the byline. I still think it could be more clear. Maybe a pipe or other punctuation after the name, then maybe some words that address the Talk or Workshop was created in 20xx (and not the human)?

dvdherron replied:
Yeah, it makes sense once you realize it means "talk first given in 20xx." Not sure what a good fix would be without being to verbose. "First given by X in 20xx"?

stacyk commented on src/scss/components/_events.scss:
```diff
@@ -1,24 +1,71 @@
+  font-size: var(--small);
+  padding: var(--gutter) 0;
+}
+
+.event-item {
+  margin-bottom: var(--double-gutter);
```
Also, I see we are showing the ordered list markers on the single event page, but not on the event list "compact" view. Is there a reason why we display them differently?

mirisuzanne commented on content/_includes/page/meta.njk:
```diff
@@ -37,7 +37,7 @@
           start=page.date,
           end=end,
-          format='range' if oss or client or event else 'long'
+          format='range' if oss or client or ('_talk' in tags) else 'long'
```
`('_talk' in tags)` might be risky if `tags` is undefined?

jgerigmeyer replied:
Hm, maybe? It doesn't seem to cause any build errors... I think 11ty ensures that `tags` always exists, it's just empty on some pages.

mirisuzanne replied:
oh also: things only show up in these lists if they have the relevant tags 😀

mirisuzanne commented on test/js/events.test.js:
```diff
@@ -84,4 +85,33 @@ describe('event filters', () => {
+      { name: 'jonny', events: [events[0]] },
+      { name: 'ed', events: [events[0], events[1], events[2]] },
+      { name: 'miriam', events: [events[1]] },
+      { name: 'oddbird', events: [events[3]] },
```
Yeah, I think it's pretty unlikely that we'd add one - but we could explicitly disallow it.

jgerigmeyer commented:
> I think this is a nice update. I don't know where to put this one, but I see the intro of the talk single on smaller screens touches the edge of the browser. But that is for all pages with this type of intro so probably not introduced in this PR. I will make a card for that.

@stacyk I made a card for that on Friday: https://trello.com/c/Te0UaZPc/200-bug-summary-text-has-no-horizontal-padding-on-narrow-screen-sizes

dvdherron commented on src/scss/components/_events.scss:
```diff
@@ -1,24 +1,75 @@
   margin: 0;
+
+  .compact-events & {
+    font: inherit;
```
```suggestion
    display: inline-block;
    font: inherit;
```

dvdherron commented on src/scss/components/_events.scss:
```diff
@@ -1,24 +1,75 @@
+  .compact-events & {
+    font: inherit;
+    font-weight: bold;
+    display: inline-block;
```
```suggestion
```

mirisuzanne commented on content/_includes/talks.macros.njk:
```diff
@@ -0,0 +1,202 @@
+    <ul class="event-links" inline-list="comma">
+      {%- for link in links -%}
+        <li class="event-link">
+          {{- utility.link_if(link.title, link.url) -}}
```
Do you know what best-practice is these days? Add sr-only text to the link, or aria-describedby connection to the heading, or something else?

mirisuzanne replied:
Removing these links from the compact view, since we can't have links in cards. But we should still solve this in the expanded view.

stacyk replied:
@mirisuzanne The expanded view is a little better since when you tab through, it reads the title of the event followed by link to slide and link to video. Since the cards didn't have the title of the event as a link (only the talk topic with the authors name after topic), it wasn't clear which slides from which event you were looking at. I think we can add the name of the conference to the screen reader text before the first link if there are two, or maybe on both. I never know what is too verbose.

stacyk replied:
Added a card for this: https://trello.com/c/g96PXZhv/201-i-want-screen-reader-friendly-links-to-event-slides-and-videos-in-cards-and-on-detail

mirisuzanne commented:
Ok, I addressed several things here:

- Date range isn't shown in the byline on talk/event pages
- Fixed the page-intro margin issue
- Added the 'past' and 'upcoming' headers to the compact lists (like before it only shows one or the other)
- Removed the 'upcoming events have a bold date'
- Removed slide/video links from compact view, since we can't do nested links in cards

Still could improve the a11y of slide/video links in the non-compact event lists.

SondraE commented:
@mirisuzanne Unless I'm miss understanding something, I'm still seeing a date in the byline on talk/event pages.
Also my read of the conversation above was that it would be helpful to add Upcoming and Past labels not only the compact view, but also to the Events section on individual talk detail pages.

mirisuzanne commented:
> Unless I'm miss understanding something, I'm still seeing a date in the byline on talk/event pages.

@SondraE - sorry, I had only removed from list view - now also removed from detail pages.

> Also my read of the conversation above was that it would be helpful to add Upcoming and Past labels not only the compact view, but also to the Events section on individual talk detail pages.

Oh, ok. I didn't read it that way, because we didn't have labels there before. We can add them, but it's more complex. In the compact view we just show one list or the other – either upcoming or past, but never both. In the full view we would need to split out two distinct lists, and show both of them. If we want to keep the single combined list that we currently have, then we need a per-event distinction rather than a list label. If we want two lists with labels, I need to make some bigger changes.

SondraE reviewed (approved):
> Also my read of the conversation above was that it would be helpful to add Upcoming and Past labels not only to the compact view, but also to the Events section on individual talk detail pages.

We talked about this at the morning meeting. Consensus was that we could create a separate card for adding Upcoming and Past labels to talk detail pages, and not tackle that in this PR.

jgerigmeyer commented on content/_includes/post.macros.njk:
```diff
@@ -349,11 +349,9 @@ params:
 
-    {% if dateformat %}
-      {% set events = post | pageEvents if ('_talk' in (post.data.tags or [])) else none %}
-      {% if not (events and events.length > 0) %}
```
@mirisuzanne Is there a reason why you were checking for events here instead of just checking if it's a talk/workshop? I didn't see any practical difference.

mirisuzanne replied:
no good reason, this change looks good to me.

jgerigmeyer commented on content/_includes/page/meta.njk:
```diff
@@ -34,13 +34,13 @@
         {{ post.by(author, collections) }}
         {{ linked | safe }}
         {{ post.client(client) }}
-        {% if tags and not ('_talk' in tags) %}
```
This was removing dates from client/oss pages (which have no `tags`).

jgerigmeyer reviewed (commented):
I find the reverse-numbering on events (both summary and detail) to be unnecessarily confusing. Can we just use `ul` there, or `ol` but without displaying the numbers?

![Screenshot 2022-12-14 at 12 35 54 PM](https://user-images.githubusercontent.com/552316/207667307-8ab7c46e-c233-4b50-8e65-e4064a7c16f5.jpg)

![Screenshot 2022-12-14 at 12 37 04 PM](https://user-images.githubusercontent.com/552316/207667309-2d8cb4b9-f9b6-4d1c-82be-b925271251af.jpg)

stacyk reviewed (approved):
Thanks for the edits! The circles are better than numbers

jgerigmeyer reviewed (approved):
@mirisuzanne Feel free to merge this if you're okay with the slight change I just pushed in https://github.com/oddbird/oddleventy/commit/70428c4b394b6b01ff7771cbec72645c9e7891cb.
